### PR TITLE
Feat/asset balances

### LINKED ISSUE #238: Show Asset Balances
As a user I want to see asset balances in my wallet.

Tasks:
- [x] Set up apollo client
- [x] Connect to one of the nodes in network configuration
- [x] Query and display balances for given key

### PR DESCRIPTION
**Pull request by mattrussell36**
Closes #238

### CODE REVIEW

File: frontend/automation/cypress/integration/wallets.js [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+  it('message signing failure', () => {
+    unlockWallet(walletName, passphrase)
+    cy.getByTestId('sign-page').click()
+    signMessage('Sign message failure')
+    authenticate('invalid')
+    cy.contains('Error').should('have.text', 'Error: wrong passphrase')
+  })
+})
+
+describe.only('wallet - assets', () => {
```
Comment (dexturr) [inline frontend/automation/cypress/integration/wallets.js:127]: .only left in

File: frontend/automation/cypress/support/commands.js [hunk 2]
```diff
@@ -8,74 +10,62 @@ Cypress.Commands.add('clean', () => {
...
 
-  return (
-    cy
-      .window()
-      // Init wallet with local vega home
-      .then(async win => {
-        const handler = win.go.backend.Handler
+Cypress.Commands.add('restoreNetwork', (handler, name) => {
+  if (!['mainnet1', 'fairground', 'custom'].includes(name)) {
+    throw new Error('Must provide fairground or mainnet1')
```
Comment (dexturr) [inline frontend/automation/cypress/support/commands.js:46]: error message does not mention custom?

File: frontend/automation/cypress/support/create-mock-config.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  URL = "console.fairground.wtf"
+  LocalPort = 1847
+`
+const DIR = './network-config'
+const dirExists = fs.existsSync(DIR)
+if (!dirExists) {
+  fs.mkdirSync(DIR)
+}
+const result = path.join(DIR, '/custom.toml')
+console.log(result)
```
Comment (dexturr) [inline frontend/automation/cypress/support/create-mock-config.js:32]:
```suggestion
```

File: frontend/src/app.tsx [hunk 4]
```diff
@@ -1,36 +1,46 @@
...
-import { useIsOnboard } from './hooks/use-is-onboard'
+import { createLogger } from './lib/logging'
 import { AppRouter } from './routes'
 
 /**
  * Initialiases the app
  */
 function AppLoader({ children }: { children: React.ReactNode }) {
-  useCheckForUpdate()
+  // useCheckForUpdate()
```
Comment (dexturr) [inline frontend/src/app.tsx:26]: Above remove or should this not be commented?

File: frontend/src/app.tsx [hunk 5]
```diff
@@ -111,3 +124,38 @@ function AppFrame({ children }: AppFrameProps) {
     </div>
   )
 }
+
+const logger = createLogger('Errorboundary')
+
+class ErrorBoundary extends Component<
```
Comment (dexturr) [inline frontend/src/app.tsx:130]: question: as we discovered recently once this is hit there is no way to recover. We do have a way to manually recover from this?

File: frontend/src/lib/apollo-client.ts [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+        definition.operation === 'subscription'
+      )
+    },
+    wsLink,
+    httpLink
+  )
+
+  const errorLink = onError(({ graphQLErrors, networkError }) => {
+    console.log(graphQLErrors)
+    console.log(networkError)
```
Comment (dexturr) [inline frontend/src/lib/apollo-client.ts:84]: should probably capture sentry errors here. Know this isn't your code and was probs my fault before hand

File: frontend/src/app.tsx [hunk 7]
```diff
@@ -1,36 +1,46 @@
...
 import { TransactionManager } from './components/transaction-manager'
 import { Colors } from './config/colors'
-import { initAppAction } from './contexts/global/global-actions'
+import {
+  initAppAction,
+  startServiceAction
+} from './contexts/global/global-actions'
 import { AppStatus, useGlobal } from './contexts/global/global-context'
 import { GlobalProvider } from './contexts/global/global-provider'
 import { useCheckForUpdate } from './hooks/use-check-for-update'
```
[the remark this answers is not in the document]
Comment (mattrussell36) [reply frontend/src/app.tsx:18]: I've re-added and called the hook in the component

File: frontend/src/app.tsx (hunk 5 above)
Comment (mattrussell36) [reply frontend/src/app.tsx:131]: Added a button reload the app.

**Review by notbot00 — APPROVED** [review]

File: frontend/src/lib/apollo-client.ts [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+        definition.kind === 'OperationDefinition' &&
+        definition.operation === 'subscription'
+      )
+    },
+    wsLink,
+    httpLink
+  )
+
+  const errorLink = onError(({ graphQLErrors, networkError }) => {
+    if (graphQLErrors?.length) {
```
Comment (dexturr) [inline frontend/src/lib/apollo-client.ts:87]: capture with sentry?

File: frontend/src/routes/onboard/onboard.tsx [hunk 9]
```diff
@@ -64,7 +70,7 @@ export function OnboardHome() {
         // Add wallets and networks to state
         dispatch({ type: 'ADD_WALLETS', wallets: onboarding.wallets })
       } else {
-        navigate(OnboardPaths.WalletCreate)
+        navigate('/onboard/wallet-create')
```
Comment (dexturr) [inline frontend/src/routes/onboard/onboard.tsx:73]: follow up (non-blocking): we use route names on all our other apps so these don't get jumped with path reordering. Perhaps we should do that here as well?

File: frontend/src/routes/onboard/onboard.tsx [hunk 10]
```diff
@@ -299,7 +305,7 @@ export function OnboardNetwork() {
   }, [dispatch, navigate])
 
   return (
-    <OnboardPanel title='Import a network' back={OnboardPaths.WalletCreate}>
+    <OnboardPanel title='Import a network' back={'/onboard/wallet-create'}>
```
Comment (dexturr) [inline frontend/src/routes/onboard/onboard.tsx:308]: nitpick

```suggestion
    <OnboardPanel title='Import a network' back='/onboard/wallet-create'>
```

File: frontend/src/routes/wallet/wallet.tsx [hunk 11]
```diff
@@ -1,7 +1,138 @@
...
+  useEffect(() => {
+    let mounted = true
+
+    const safeSetClient = (client: ApolloClient<any> | null | false) => {
+      if (mounted) {
+        setClient(client)
+      }
+    }
+
+    ;(async () => {
```
Comment (dexturr) [inline frontend/src/routes/wallet/wallet.tsx:34]: weird semi?

**Review by dexturr — DISMISSED** [review]
Probably missed some stuff as so big, but looks good to me 👍 nicely done dude

File: frontend/src/routes/onboard/onboard.tsx (hunk 9 above)
Comment (mattrussell36) [reply frontend/src/routes/onboard/onboard.tsx:73]: Lets follow up!

**Pull request merged**